src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -81,35 +94,54 @@ protected override async Task HandleWebResourceRequest(CoreWebView2WebResourceRe
 				var headerString = GetHeaderString(headers);
 				eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(ms, statusCode, statusMessage, headerString);
 			}
-			else
+			else if (new Uri(requestUri) is Uri uri && new Uri(AppOrigin).IsBaseOf(uri))
 			{
-				// Next, try to go through WinUI Storage to find a static web asset
-				var uri = new Uri(requestUri);
-				if (new Uri(AppOrigin).IsBaseOf(uri))
-				{
-					var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
-
-					// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
-					// in which case we should allow falling back on the host page.
-					if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
-					{
-						relativePath = _hostPageRelativePath;
-					}
-					relativePath = Path.Combine(_contentRootDir, relativePath.Replace('/', '\\'));
+				var relativePath = new Uri(AppOrigin).MakeRelativeUri(uri).ToString();
 
+				// If the path does not end in a file extension (or is empty), it's most likely referring to a page,
+				// in which case we should allow falling back on the host page.
+				if (allowFallbackOnHostPage && !Path.HasExtension(relativePath))
+				{
+					relativePath = _hostPageRelativePath;
+				}
+				relativePath = Path.Combine(_contentRootDir, relativePath.Replace('/', '\\'));
+				statusCode = 200;
+				statusMessage = "OK";
+				var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
+				headers = StaticContentProvider.GetResponseHeaders(contentType);
+				var headerString = GetHeaderString(headers);
+				IRandomAccessStream? stream = null;
+				if (_isPackagedApp)
+				{
 					var winUIItem = await Package.Current.InstalledLocation.TryGetItemAsync(relativePath);
 					if (winUIItem != null)
 					{
-						statusCode = 200;
-						statusMessage = "OK";
-						var contentType = StaticContentProvider.GetResponseContentTypeOrDefault(relativePath);
-						headers = StaticContentProvider.GetResponseHeaders(contentType);
-						var headerString = GetHeaderString(headers);
-						var stream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
-
-						eventArgs.Response = _coreWebView2Environment!.CreateWebResourceResponse(stream.AsRandomAccessStream(), statusCode, statusMessage, headerString);
+						var contentStream = await Package.Current.InstalledLocation.OpenStreamForReadAsync(relativePath);
```

Comment:
I think the logic to load a stream may exist in the FilesystemUtils helpers in essentials too. I think we are using that for loading fonts and whatnot. Might be worth using that to avoid duplication.

Reply:
This deals with a bunch of WebView2 specific issues, so I prefer to keep it this way. I'm open to update this code in the future if we can, but I want to avoid changes at this point to avoid surprises.